File: Project3_Python/Plants/Guarana.py
import World
from Plant import Plant
from Organism import OrganismsEnum


class Guarana(Plant):
    def __init__(self, world, position):
        super().__init__(world, position)
        self.type_of_organism = OrganismsEnum.Guarana

    def collision(self, attacker_organism):
        attacker_organism.put_into_world()
        if self.strength > attacker_organism.get_strength():
            self.put_into_world()
            if attacker_organism.get_type_of_organism() == OrganismsEnum.Human:
                self._world.delete_human()
                return
            self._world.remove_organism_from_list(attacker_organism)
            self._world.add_comment(str(attacker_organism) + " ate " + str(self) + " and died.")
        else:
            self._world.remove_organism_from_list(self)
            attacker_organism.increase_strength(3)
            self._world.add_comment(str(attacker_organism) + " ate " + str(self) + " and got stronger.")

    def __str__(self):
        return "Guarana"

File: Project2_Java/src/pl/pg/oop/project2/organisms/plants/Guarana.java
package pl.pg.oop.project2.organisms.plants;

import pl.pg.oop.project2.organisms.Organism;
import pl.pg.oop.project2.organisms.OrganismsEnum;
import pl.pg.oop.project2.organisms.Plant;
import pl.pg.oop.project2.organisms.World;

import java.awt.*;

public class Guarana extends Plant {
    public Guarana(World world, Point position) {
        super(world, position);
        this.TypeOfOrganism = OrganismsEnum.GUARANA;
    }

    @Override
    public void collision(Organism AttackerOrganism) {
        AttackerOrganism.PutIntoWorld();
        if(strength > AttackerOrganism.getStrength()) {
            PutIntoWorld();
            if(AttackerOrganism.getTypeOfOrganism() == OrganismsEnum.HUMAN) {
                world.DeleteHuman();
                return;
            }
            world.RemoveOrganismFromList(AttackerOrganism);
            world.addComment(AttackerOrganism + " ate " + this + " and died.");
        }
        else {
            world.RemoveOrganismFromList(this);
            AttackerOrganism.increaseStrength(3);
            world.addComment(AttackerOrganism + " ate " + this + " and got stronger.");
        }
    }

    @Override
    public String toString() {
        return "Guarana";
    }
}
